ConfigurationManager/Interface/KeyboardShortcut.cs

```diff
@@ -125,48 +99,61 @@ public static KeyboardShortcut Deserialize(string str)
 
         public string Serialize()
         {
-            return $"{(int)Key} {(Control ? 1 : 0)} {(Alt ? 1 : 0)} {(Shift ? 1 : 0)}";
+            return string.Join(" ", allKeys.Select(x => x.ToString()).ToArray());
         }
 
         /// <summary>
         /// Check if the main key was just pressed (Input.GetKeyDown), and specified modifier keys are all pressed
         /// </summary>
         public bool IsDown()
         {
-            return Input.GetKeyDown(Key) && ModifierKeyTest();
+            if (MainKey == KeyCode.None) return false;
+
+            return Input.GetKeyDown(MainKey) && ModifierKeyTest();
         }
 
         /// <summary>
         /// Check if the main key is currently held down (Input.GetKey), and specified modifier keys are all pressed
         /// </summary>
         public bool IsPressed()
         {
-            return Input.GetKey(Key) && ModifierKeyTest();
+            if (MainKey == KeyCode.None) return false;
+
+            return Input.GetKey(MainKey) && ModifierKeyTest();
         }
 
         /// <summary>
         /// Check if the main key was just lifted (Input.GetKeyUp), and specified modifier keys are all pressed.
         /// </summary>
         public bool IsUp()
         {
-            return Input.GetKeyUp(Key) && ModifierKeyTest();
+            if (MainKey == KeyCode.None) return false;
+
+            return Input.GetKeyUp(MainKey) && ModifierKeyTest();
         }
 
         private bool ModifierKeyTest()
         {
-            if (Key == KeyCode.None)
-                return false;
-
-            if (Control && !Input.GetKey(KeyCode.LeftControl) && !Input.GetKey(KeyCode.RightControl))
-                return false;
-
-            if (Alt && !Input.GetKey(KeyCode.LeftAlt) && !Input.GetKey(KeyCode.RightAlt))
-                return false;
+            return AllKeys.All(c =>
```

Comment:
Nice. As I mentioned, the change does indeed reduce the amount of code.

Any reason to poll *all* keys?
Why not just iterate an shortcircuit on defined keys?

For instance:
```csharp
for(var key in allKeys) 
    if(!Input.GetKey(key))
        return false;
return true;
```
You can optimise this a bit to skip the main key, but to be honest I'd rather assume that the *last* key is the "main key" (as it is usually notated; after all, we usually write <kbd>Ctrl</kbd>+<kbd>V</kbd>, not <kbd>V</kbd>+<kbd>Ctrl</kbd>). This assumption would in fact allow for a much more concise (and optimised) version:

```csharp
for(int i = 0; i < allKeys.Length - 1; i++)
    if(!Input.GetKey(allKeys[i]))
        return false;
return true;
```

Reply:
I need to poll all keys to avoid triggering X + Shift shortcut when user presses X + Shift + Control or any other combination that just happens to include X as main key and Shift as one of the modifiers.

I always think of first key being the main, maybe it could be made clearer somehow.

Reply:
Fair enough then, even though I'm quite sure no one would use already-bound keys.

Have you tested how the performance is affected from you iterating through all keys? Calling `Contains` on an array/list is O(n) operation, so essentially we're looking for a O(n^2) as the worst case modifier lookup -- though O(n*m), where m = number of all bindable keys, is more realistic.

This is one of the reasons I ask why you poll it all and whether it's actually worth having two already bound keys bindable again.